File: PreferenceFragmentCompat/app/src/main/java/junseong/k/myapplication/MainActivity.java
package junseong.k.myapplication;

import androidx.appcompat.app.AppCompatActivity;
import androidx.preference.PreferenceFragmentCompat;
import androidx.preference.PreferenceManager;

import android.content.Intent;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity {
    public static final String TARGET_SETTING_PAGE="target";
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        SettingPreferenceFragment settingPreferenceFragment=new SettingPreferenceFragment();
        Intent intent=getIntent();
        if(intent!=null){
            String rootkey=intent.getStringExtra(TARGET_SETTING_PAGE);
            if(rootkey!=null){
                Bundle args=new Bundle();
                args.putString(PreferenceFragmentCompat.ARG_PREFERENCE_ROOT,rootkey);
                settingPreferenceFragment.setArguments(args);
            }
        }
        getSupportFragmentManager().beginTransaction().replace(android.R.id.content,settingPreferenceFragment,null).commit();
    }
}

File: AutoCompleteTextView(Kotlin)/app/src/main/java/junseong/kotlin/autocompletetextview/MainActivity.kt
package junseong.kotlin.autocompletetextview

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.ArrayAdapter
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    val data1= arrayOf("abcd","abca","abcb","abcc","bbaa","bbab","bcab","bdab")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val apdater1=ArrayAdapter(this,android.R.layout.simple_dropdown_item_1line,data1)
        autotext.setAdapter(apdater1)
        button.setOnClickListener {
            text1.text=autotext.text
        }
       // autotext.setOnItemClickListener(listner1)
        autotext.setOnItemClickListener { parent, view, position, l ->
            text2.text="${data1[position]} 항목을 클릭스 "
        }
    }
    val listner1=object:AdapterView.OnItemClickListener{
        override fun onItemClick(p0: AdapterView<*>?, p1: View?, p2: Int, p3: Long) {
            text2.text="${data1[p2]} 항목을 클릭하였습니다"
        }
    }
}

